usr/src/uts/common/fs/zfs/vdev.c

```diff
@@ -2449,13 +2450,12 @@ vdev_online(spa_t *spa, uint64_t guid, uint64_t flags, vdev_state_t *newstate)
 	if (!vd->vdev_ops->vdev_op_leaf)
 		return (spa_vdev_state_exit(spa, NULL, ENOTSUP));
 
-	postevent =
-	    (vd->vdev_offline == B_TRUE || vd->vdev_tmpoffline == B_TRUE) ?
-	    B_TRUE : B_FALSE;
+	wasoffline = (vd->vdev_offline || vd->vdev_tmpoffline);
+	oldstate = vd->vdev_state;
 
 	tvd = vd->vdev_top;
-	vd->vdev_offline = B_FALSE;
-	vd->vdev_tmpoffline = B_FALSE;
+	vd->vdev_offline = 0ULL;
+	vd->vdev_tmpoffline = 0ULL;
```

Comment:
You'll see the vdev_tmpoffline is already declared a boolean_t so the only required change is to vdev_offline. I don't believe there is any reason for them being uint64_ts since its value can only be 0 or 1. Long ago, we may have actually had this to make nvlist_add_uint64() happen when we add this value to the label but that code has also changed to only pass B_TRUE.